app/controllers/support_controller.rb

```diff
@@ -1,3 +1,50 @@
 class SupportController < ApplicationController
-  def support; end
+  def support
+    @support_form = SupportForm.new
+  end
+
+  def new
+    @support_form = SupportForm.new(support_form_params)
+
+    if @support_form.invalid?
+      render :support and return
+    end
+
+    case @support_form.i_need_help_with.to_sym
```

Comment:
Why are we doing routing here? Can we not plonk these into the routes file? https://rubyhero.dev/rails-routes-less-known-features#heading-more-complex-redirection-logic

Would be preferable as well to not redirect but route directly

Reply:
> Why are we doing routing here? Can we not plonk these into the routes file?

I did not know that you could do that 😂 

> Would be preferable as well to not redirect but route directly

I considered that initially, but then I decided against it, because I thought it would make the routing a bit more confusing in the scenario where a user makes an error on the second page of the form (for example forgetting to put their name). Maybe the technique you've suggested could reduce some of the complexity in the controller?

But to be honest I found the more I thought about it when I was thinking about it earlier, it seemed weird to have multiple different pages at the same path, it almost felt trying to emulate a single page application or something. What would happen with the browser history and the back button? So I ended up leaning towards redirecting being more appropriate.

What was your thinking behind your preference for routing directly? Are there strong reasons to avoid the redirect?

Reply:
You generally want to send the user to a page as quickly as possible. I wouldn't be so bothered if this was on the admin app, but as this is user facing we can't predict what their experience will be like. An additional request might add seconds to their journey.

> it seemed weird to have multiple different pages at the same path

100% agree. Ideally the route can check the params and respond with the appropriate controller action according to the values, without a redirect and without having to do it in the controller itself.